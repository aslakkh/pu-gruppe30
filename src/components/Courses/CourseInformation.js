import React, { Component } from 'react'
import {ListGroup, ListGroupItem, Panel} from 'react-bootstrap'
import { getCourseMessages } from '../../helpers/auth'
import firebase from 'firebase';
import './CourseInformation.css';


export default class CourseInformation extends Component{
    /*get information from Firebase (course/messages/something)
    display list of five last messages
    */

    constructor(props){
        super(props);
        this.state={
            courseID: this.props.courseID,
        }
    }

    componentWillReceiveProps(nextProps){
        var messages = this.getMessages();
        this.setState({
            courseID: nextProps.courseID,
            messages: messages,
        })
    }
    //retreive messages from firebase, set component state
    componentDidMount(){
        let that = this;
        var courseRef = firebase.database().ref().child('courses/'+this.state.courseID+'/Messages');
        courseRef.on('value', snap => {
            that.setState({
                messages: snap.val(),
            });
        });
        
    }

    getMessages(){
        let that = this;
        var courseRef = firebase.database().ref().child('courses/'+this.state.courseID+'/Messages');
        var messages;
        courseRef.on('value', snap => {
            messages = snap.val();
        });
        return messages;
    }

    msToDateString(dateInMs){
        let d = new Date(parseInt(dateInMs));
        return (d.toDateString());
    }


    render(){
        //render based on if course has messages
        console.log("COURSE INFO: " + this.state.courseID);
        if(this.state.messages === null || this.state.messages === undefined){
            return(
                <div>
                <Panel bsStyle="info" collapsible defaultExpanded header="Messages from course staff" className="PanelHeader">
                    <ListGroup fill>
                        <ListGroupItem><h5>No messages to be displayed.</h5></ListGroupItem>
                    </ListGroup>
                </Panel>
            </div>
            )
        }
        else{
            return(
            <div>
                <Panel bsStyle="info" collapsible defaultExpanded header="Messages from course staff" className="PanelHeader">
                    <ListGroup fill>
                        {Object.keys(this.state.messages).map((key) => {
                                    return <ListGroupItem key={key} header={this.state.messages[key].Message} className="MessagesListItem">
                                                <h5>{this.msToDateString(key)}</h5>
                                            </ListGroupItem>
                                })} 
                    </ListGroup>
                </Panel>
            </div>
        );
        }
        

    }
}