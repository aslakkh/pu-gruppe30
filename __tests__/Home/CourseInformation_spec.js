/**
 * Created by jan on 25/04/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import CourseInformation from '../../src/components/Courses/CourseInformation.js';
import TestUtils from "react-addons-test-utils"
import {shallow} from 'enzyme';

const courses={
    TES1000:{active: true,
        goals: {dailyGoal:{active: false,
            timeSet: 0,
            timeSpent: 0,
            value: 0},
            monthlyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0},
            weeklyGoal:
                {active:false,
                    timeSet:0,
                    timeSpent:0,
                    value:0}

        }
    }};
const coursesNull=null;
describe("CourseInformation", function () {
    let renderer2;
    let component2;
    renderer2 = TestUtils.createRenderer();
    renderer2.render(
        <CourseInformation courseID={"TES1000"}/>
    );
    component2 = renderer2.getRenderOutput();
    it("One course", () => {
        expect(component2).toMatchSnapshot();
    });

    renderer2.render(
        <CourseInformation courseID={"TDT4100"}/>
    );
    component2 = renderer2.getRenderOutput();
    it("Different course", () => {
        expect(component2).toMatchSnapshot();
    });
});