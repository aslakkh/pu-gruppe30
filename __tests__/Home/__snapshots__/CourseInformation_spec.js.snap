// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseInformation Different course 1`] = `
<div>
  <Panel
    bsClass="panel"
    bsStyle="info"
    className="PanelHeader"
    collapsible={true}
    defaultExpanded={true}
    header="Messages from course staff"
  >
    <ListGroup
      bsClass="list-group"
      fill={true}
    >
      <ListGroupItem
        bsClass="list-group-item"
        listItem={false}
      >
        <h5>
          No messages to be displayed.
        </h5>
      </ListGroupItem>
    </ListGroup>
  </Panel>
</div>
`;

exports[`CourseInformation One course 1`] = `
<div>
  <Panel
    bsClass="panel"
    bsStyle="info"
    className="PanelHeader"
    collapsible={true}
    defaultExpanded={true}
    header="Messages from course staff"
  >
    <ListGroup
      bsClass="list-group"
      fill={true}
    >
      <ListGroupItem
        bsClass="list-group-item"
        listItem={false}
      >
        <h5>
          No messages to be displayed.
        </h5>
      </ListGroupItem>
    </ListGroup>
  </Panel>
</div>
`;
